packages/slate-react/src/plugin/react-editor.ts

```diff
@@ -463,6 +464,15 @@ export const ReactEditor = {
       let leafNode = parentNode.closest('[data-slate-leaf]')
       let domNode: DOMElement | null = null
 
+      // The parent of TEXT_NODE must have `date-slate-string`.
+      if (
+        !voidNode &&
+        isDOMText(nearestNode) &&
+        !parentNode.hasAttribute('data-slate-string')
+      ) {
+        parentNode.removeChild(nearestNode)
```

Comment:
I'm not sure I think the right answer is to simply remove the just inserted text. Would it not be better to wrap it in a valid parent with a transform/normalization rule instead?